coalib/results/Diff.py

```diff
@@ -351,14 +351,17 @@ def __bool__(self):
     def delete_line(self, line_nr):
         """
         Mark the given line nr as deleted. The first line is line number 1.
+        Raises an error if line nr doesn't exist in the diff.
         """
+        line_nr = 0 if (len(self._file) < line_nr) else line_nr
         linediff = self._get_change(line_nr)
         linediff.delete = True
         self._changes[line_nr] = linediff
 
     def delete_lines(self, line_nr_start, line_nr_end):
         """
-        Delete lines in a specified range, inclusively.
+        Delete lines in a specified range, inclusively. The range must be
+        valid, i.e. lines must exist in diff, else an error message is raised.
```

Comment:
`delete_lines` calls `delete_line` which calls `_get_change`. and error is raised in `_get_change`